### PR TITLE
Split RestCallback in 3 interfaces

### PR DESCRIPTION
**Pull request by meriouma**

### CODE REVIEW

**Comment by olafleur** [comment]
LGTM

File: dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java [hunk 1]
```diff
@@ -35,33 +40,65 @@
...
+    public ResourceDelegate<R> withDelegatingDispatchRequest(DelegatingDispatchRequest delegatingDispatchRequest) {
+        AbstractResourceDelegate<R> delegate = createCopy();
         delegate.delegatingDispatchRequest = delegatingDispatchRequest;
 
         return delegate;
     }
 
     @Override
-    public T withoutCallback() {
+    public R withoutCallback() {
```
Comment (Chris-V) [inline dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java]: I think we can safely remove that method. wrapCallback will have the same effect if nothing is specified.

File: dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java [hunk 2]
```diff
@@ -75,16 +112,44 @@ public T withCallback(RestCallback<?> callback) {
...
-        AbstractResourceDelegate<T> delegate = newInstance();
+    private AbstractResourceDelegate<R> createCopy() {
+        AbstractResourceDelegate<R> delegate = newInstance();
         copyFields(delegate);
 
         return delegate;
     }
+
+    private RestCallback<?> wrapCallbacks() {
+        return new RestCallback() {
```
Comment (Chris-V) [inline dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java]: Maybe extract it to a package-local class

**Comment by Chris-V** [comment]
LGTM

File: dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java (hunk 1 above)
Comment (meriouma) [inline dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java]: I guess it depends if we want to keep the syntax `delegate.withoutCallback().doSomething()` vs keeping only `delegate.call().doSomething()`

Comment (Chris-V) [inline dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java]: Does it add much to the readability?

Comment (meriouma) [inline dispatch-rest-delegates/src/main/java/com/gwtplatform/dispatch/rest/delegates/client/AbstractResourceDelegate.java]: Meh, I guess not that much

**Pull request merged**